src/main.rs

```diff
@@ -197,51 +197,57 @@ fn decode(input_path: &Path, restored_path: &Path) -> Result<(), parser::Restore
 }
 
 fn main() -> Result<(), parser::RestoreError> {
-    let matches = App::new("qrxfil")
-        .version("0.1")
-        .about("Transfer/backup files as a sequence of QR codes")
-        .author("Jb DOYON") // And authors
-        .subcommand(
-            SubCommand::with_name("exfil")
-                .about("Generates QR code sequence from file")
-                .arg(
-                    Arg::with_name("input") // And their own arguments
-                        .help("The input file to split into QR codes")
-                        .index(1)
-                        .required(true),
-                )
-                .arg(
-                    Arg::with_name("output_folder")
-                        .help("The output folder to generate codes into")
-                        .index(2)
-                        .required(true),
-                ),
-        )
-	.subcommand(
-	                SubCommand::with_name("restore")
-                .about("Decodes encoded strings back into file")
-                .arg(
-                    Arg::with_name("encoded_input") // And their own arguments
-                        .help("The input file with newline-delimited QR strings")
-                        .index(1)
-                        .required(true),
-                )
-                .arg(
-                    Arg::with_name("output_file")
-                        .help("The output file to restore into")
-                        .index(2)
-                        .required(true),
-                ),
-
-	)
-        .get_matches();
-
-    // // You can check if a subcommand was used like normal
-    // let matches_exfil = match matches.subcommand_matches("exfil") {
-    //     Some(i) => i,
-    //     None => panic!("Exfil alone is implemented"),
-    // };
+    let matches = get_args();
+    let result = run(&matches);
+    if let Err(ref e) = result {
+        eprintln!("{}", e);
+    }
+
+    result
+}
+
+fn get_args() -> ArgMatches<'static> {
```

Comment:
Why didn't I think of that!? Refactoring args to func is so obvious.

Well, I know I wouldn't have done the static bit. What's that about? also the `'` for static. weiiird

Reply:
`'static` is a lifetime, that's what the leading single quote signifies.

Because all the arguments that go into building the args are static (ie will live as long as the whole program) then the result has a static lifetime.

You could imagine a situation where one of the inputs to a Clap function had a finite lifetime (say `'a`), then the output would also have a lifetime of `'a`, since it contains a reference to it's input, and it must not live longer than the reference